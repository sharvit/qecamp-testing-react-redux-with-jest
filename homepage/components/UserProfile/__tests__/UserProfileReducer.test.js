import { testReducerSnapshotWithFixtures } from '../../../../testHelpers';

import {
  USER_PROFILE_UPDATE_AVATAR,
  USER_PROFILE_UPDATE_POSTS,
  USER_PROFILE_UPDATE_PHOTOS,
} from '../UserProfileConstants';
import reducer from '../UserProfileReducer';

const fixtures = {
  'should return the initial state': {},
  'should handle USER_PROFILE_UPDATE_AVATAR': {
    action: {
      type: USER_PROFILE_UPDATE_AVATAR,
      payload: true,
    },
  },
  'should handle USER_PROFILE_UPDATE_POSTS': {
    action: {
      type: USER_PROFILE_UPDATE_POSTS,
      payload: true,
    },
  },
  'should handle USER_PROFILE_UPDATE_PHOTOS': {
    action: {
      type: USER_PROFILE_UPDATE_PHOTOS,
      payload: true,
    },
  },
};

describe('UserProfile - Reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
