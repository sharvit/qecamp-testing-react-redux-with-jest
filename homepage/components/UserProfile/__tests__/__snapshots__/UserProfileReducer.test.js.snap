// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile - Reducer should return the initial state 1`] = `
Object {
  "showAvatar": false,
  "showPhotos": false,
  "showPosts": false,
}
`;

exports[`UserProfile - Reducer should handle USER_PROFILE_UPDATE_AVATAR 1`] = `
Object {
  "showAvatar": true,
  "showPhotos": false,
  "showPosts": false,
}
`;

exports[`UserProfile - Reducer should handle USER_PROFILE_UPDATE_PHOTOS 1`] = `
Object {
  "showAvatar": false,
  "showPhotos": true,
  "showPosts": false,
}
`;

exports[`UserProfile - Reducer should handle USER_PROFILE_UPDATE_POSTS 1`] = `
Object {
  "showAvatar": false,
  "showPhotos": false,
  "showPosts": true,
}
`;
